### PR TITLE
docs(samples): adds routine and user agent samples

### PR DESCRIPTION
**Pull request by steffnay**
- Adds samples for get/list/update/delete routines 

- Adds sample for setting provided user agent

- Updates samples dataset tests to delete stale datasets

- Refactors samples tests resource naming

- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

🦕

### CODE REVIEW

File: samples/createRoutine.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      },
+    };
+
+    // Make API call
+    [routine] = await routine.create(config);
+
+    console.log(`Routine ${routineId} created.`);
+  }
+  // [END bigquery_create_routine]
+  createRoutine();
```
Comment (JustinBeckwith) [inline samples/createRoutine.js:62]: This needs to be inside the snippet region tag

File: samples/createRoutineDDL.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    const [job] = await bigquery.createQueryJob(options);
+    console.log(`Job ${job.id} started.`);
+
+    // Wait for the query to finish
+    await job.getQueryResults();
+
+    console.log(`Routine ${routineId} created.`);
+  }
+  // [END bigquery_create_routine_ddl]
+  createRoutineDDL();
```
Comment (JustinBeckwith) [inline samples/createRoutineDDL.js:50]: Here and below, need these inside the region tags

File: samples/getRoutine.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * TODO(developer): Uncomment the following lines before running the sample.
+     */
+    // const datasetId = 'my_dataset';
+    // const routineId = 'my_routine';
+
+    const dataset = bigquery.dataset(datasetId);
+
+    // Create routine reference
+    let routine = dataset.routine(routineId);
```
Comment (JustinBeckwith) [inline samples/getRoutine.js:38]: Could we shorten this to:
```js
const [routine] = await dataset.routine(routineId).get();
```

File: samples/setUserAgent.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+'use strict';
+
+function main() {
+  // [START bigquery_set_user_agent]
+  // Import the Google Cloud client library
+  const {BigQuery} = require('@google-cloud/bigquery');
+  // Create a client and set the user agent
+  const bigquery = new BigQuery({userAgent: 'my-user-agent'});
+
+  async function setUserAgent() {
```
Comment (JustinBeckwith) [inline samples/setUserAgent.js:24]: So this sample has no `await` in it - so no need to have the wrapping async function!  You can drop the `async function setUserAgent` wrapping.

File: samples/test/routines.test.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+const newRoutineId = generateUuid();
+
+const bigquery = new BigQuery();
+
+describe('Routines', () => {
+  after(async () => {
+    await bigquery
+      .dataset(datasetId)
+      .delete({force: true})
+      .catch(console.warn);
```
Comment (JustinBeckwith) [inline samples/test/routines.test.js:38]: Do we have any concerns with leaking datasets here?  I'm ok with the `console.warn`, it's just unlikely it will ever be seen unless running locally.  What's the chance of leakage?

Comment (steffnay) [reply samples/test/routines.test.js:38]: @JustinBeckwith I think there's a really low chance of this failing. But we would leak datasets if the tests were stopped mid-run. An option to have a failed delete retried at some point could be to add labels to these testing datasets, then add logic to the before hook to find and delete any stale datasets with the label. This is the pattern used in the system tests and I think it makes sense to just add it to all the samples tests too.

Comment (steffnay) [reply samples/test/routines.test.js:42]: I added `deleteDatasets()` to the datasets test before hook, so leaked datasets should be cleaned up on subsequent runs.

File: samples/test/datasets.test.js [hunk 6]
```diff
@@ -105,4 +111,34 @@ describe('Datasets', () => {
     const [exists] = await bigquery.dataset(datasetId).exists();
     assert.strictEqual(exists, false);
   });
+
+  // Only delete a resource if it is older than 24 hours. That will prevent
+  // collisions with parallel CI test runs.
+  function isResourceStale(creationTime) {
+    const oneDayMs = 86400000;
```
Comment (JustinBeckwith) [inline samples/test/datasets.test.js:118]: Non-blocking feedback, I like to spell this out as `24 * 60 * 60 * 1000` to make it easier to reason about

File: samples/test/datasets.test.js [hunk 7]
```diff
@@ -105,4 +111,34 @@ describe('Datasets', () => {
...
+    const now = new Date();
```
Comment (JustinBeckwith) [inline samples/test/datasets.test.js:119]: Another approach you could take here is to use `Date.now()`.  If `creationTime` is  a "milliseconds since bleh" kind of value:
```js
return Date.now() - creationTime >= oneDayMs
```

**Review by JustinBeckwith — APPROVED** [review]
